etl-pipeline/scripts/GRIN_initial_scrape.py

```diff
@@ -0,0 +1,134 @@
+import logging
+from typing import List,Iterator
+from google.auth.transport.requests import (
+    AuthorizedSession,
+)
+from model import GRINState, GRINStatus, Record,FRBRStatus
+from managers import DBManager
+from google.oauth2.service_account import Credentials
+import json
+from services.ssm_service import SSMService
+from uuid import uuid4
+import argparse
+import os
+
+
+DEFAULT_BASE_URL = "https://books.google.com/libraries/NYPL"
+DEFAULT_CHUNK_SIZE = 5000
+
+logging.basicConfig(
+    filename="GRIN_initial_scrape.log",
+    level=logging.INFO,  
+    format='%(asctime)s - %(levelname)s - %(message)s' 
+)
+
+class AuthSession(object):
+    """Eventually, this will a service, we will use this mock code for now"""
+    def __init__(self):
+        self.creds = self.load_creds()
+        self.session = AuthorizedSession(self.creds)
+
+    def get_session(self):
+        return self.session
+
+    def load_creds(self):
+        ssm_service = SSMService()
+        service_account_file = ssm_service.get_parameter('grin-auth')
+        service_account_info = json.loads(service_account_file)
+
+        scopes = [
+            'openid',
+            'https://www.googleapis.com/auth/userinfo.email',
+            'https://www.googleapis.com/auth/userinfo.profile'
+        ]
+
+        creds = Credentials.from_service_account_info(
+            service_account_info, scopes=scopes)
+        return creds
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--url", default=DEFAULT_BASE_URL)
+    parser.add_argument("--size", default=DEFAULT_CHUNK_SIZE)
+    args = parser.parse_args()
+
+    auth = AuthSession()
+    authed_session = auth.get_session()
+
+    db_manager = DBManager(
+        user=os.environ.get("POSTGRES_USER", None),
+        pswd=os.environ.get("POSTGRES_PSWD", None),
+        host=os.environ.get("POSTGRES_HOST", None),
+        port=os.environ.get("POSTGRES_PORT", None),
+        db=os.environ.get("POSTGRES_NAME", None),
+    )
+    db_manager.create_session()
+
+    url = f"{args.url}/_all_books?book_state=NEW&book_state=PREVIOUSLY_DOWNLOADED&format=text"
+    logging.info(f"Scrapping url: {url}")
+
+    response = authed_session.get(url, timeout=600)
+    response.raise_for_status()
+   
+    barcodes = response.content.decode("utf8").strip().split("\n")
+
+    if len(barcodes) > 0: 
+        insert_into_db(barcodes=barcodes, db_manager=db_manager, chunk_size=args.size)
+    else:
+        logging.info("Zero record found")
+
+def insert_into_db(barcodes: List[str], db_manager: DBManager, chunk_size: int):
+    logging.info(f"Processing {len(barcodes)} barcodes")
+
+    for chuncked_barcodes in chunk(iter(barcodes), chunk_size):
+        new_records: List[Record] = []
+        for barcode in chuncked_barcodes:
+            _barcode_id = f"NYPL_{barcode}"
+            new_records.append(
+                Record(
+                    uuid=uuid4(),
+                    frbr_status=FRBRStatus.TODO.value,
```

Comment:
These two fields are required, I am using `to_do` as the default value but please LMK if I shouldn't be using this.